Add information about markdowns to the FAQ (#506)
hashicorp/terraform-docs-common: website/docs/registry/faq.mdx

If you have already released a version of your provider and are unhappy with its documentation, [fix the documentation issue in your repository and include it in a new release](#how-do-i-correct-bugs-in-published-versions).

### Why is my documentation not rendered like it is in Github?

> We make every effort to support GitHub Markdown syntax in the registry documentation, but there may be instances where compatibility limitations arise. To ensure consistent rendering in the Public Registry, we recommend using syntax widely supported across various Markdown flavors. Use the [documentation preview tool](https://registry.terraform.io/tools/doc-preview) to view how the Terraform Registry will format your documentation.

Reviewer: We try to only capitalize products in the docs, and another teeny suggestion for conciseness:
```suggestion
We make every effort to support GitHub Markdown syntax in the registry documentation, but occasionally compatibility limitations arise. To ensure consistent rendering in the public registry, we recommend using syntax widely supported across various Markdown flavors. Use the [documentation preview tool](https://registry.terraform.io/tools/doc-preview) to view how the Terraform Registry will format your documentation.
```

## Can I regain ownership of my artifact after losing access to my GitHub account?

For support, please [reach out to us via email](/terraform/registry#getting-help).

## Documentation Markdown Support

### Why is my documentation not rendered like it is in Github?

> We make every effort to support GitHub Markdown syntax, but there are some limitations. To ensure consistent rendering across platforms, we recommend using generic Markdown syntax whenever possible.

Reviewer 2: Is there any link we can provide to what "generic Markdown syntax" is / means?

Reviewer 2: ```suggestion
We make every effort to support GitHub Markdown syntax, but there are some limitations. To ensure consistent rendering in the Public Registry, we recommend using generic Markdown syntax whenever possible.
```

> ### Why are referenced links not rendered?

Reviewer 2: It feels a little weird to me to just have this one example of a specific case where rendering won't work. I do understand the intent is to add to this as more divergences from GitHub syntax come to our attention. But are there any other cases we are aware of right now that we could add to make this feel a bit less odd?

## Can I regain ownership of my artifact after losing access to my GitHub account?

For support, please [reach out to us via email](/terraform/registry#getting-help).

> 

Reviewer: **Suggestion:** what if instead of adding a new question here, you instead add these sentences to the question ["How do I fix provider documentation issues?"](https://terraform-docs-common-bcjkdh2u9-hashicorp.vercel.app/terraform/registry/faq#how-do-i-fix-provider-documentation-issues) above. It feels like this question is related, and then all the information about rendering docs is grouped together at the top.

Author: This makes sense to me. Instead of it just being a sentence, can I add the h3 under the "How do I fix provider documentation issues?" question?

Reviewer: I'd recommend double-checking the look of it in Vercel after making the change; but I think that makes sense!

> ## Documentation Markdown Support

Reviewer: If you don't like my above suggestion, then I'd recommend removing this header in favor of converting the h3 below into an h2. Then the question would match the rest of this page's format:

```suggestion
```

(for visual reference here is the right-hand TOC )
![Screenshot 2023-12-20 at 11 00 11 AM](https://github.com/hashicorp/terraform-docs-common/assets/32436232/c175c136-6429-45dc-a48d-ed8825461b69)

> ### Why is my documentation not rendered like it is in Github?

Reviewer: ```suggestion
## Why is my documentation not rendered like it is in Github?
```

> We make every effort to support GitHub Markdown syntax in the registry documentation, but there may be instances where compatibility limitations arise. To ensure consistent rendering in the Public Registry, we recommend using syntax widely supported across various Markdown flavors. You can test out your documentation in the registry docs preview tool at https://registry.terraform.io/tools/doc-preview.

Reviewer: I'd recommend using an inline link for better readability. 
```suggestion
We make every effort to support GitHub Markdown syntax in the registry documentation, but there may be instances where compatibility limitations arise. To ensure consistent rendering in the Public Registry, we recommend using syntax widely supported across various Markdown flavors. Use the [documentation preview tool](https://registry.terraform.io/tools/doc-preview) to view how the Terraform Registry will format your documentation.
```

### Why is my documentation not rendered like it is in Github?

> We make every effort to support GitHub Markdown syntax in the registry documentation, but there may be instances where compatibility limitations arise. To ensure consistent rendering in the Public Registry, we recommend using Markdown syntax widely supported in different Markdown flavors. You can test out your documentation in the registry docs preview tool at https://registry.terraform.io/tools/doc-preview.

Reviewer 2: I tried out the spaces-before-reference-links problem in the doc-preview tool... and it works. What do you make of this?

<img width="632" alt="Screenshot 2023-12-06 at 12 44 29 PM" src="https://github.com/hashicorp/terraform-docs-common/assets/515424/0bd32b37-a2c4-41eb-b6a6-88c909331a40">

Author: try making it 4 spaces
```
this is a [link to google][1]
    [1]: www.google.com`
```
